### PR TITLE
[exporter/mezmo] Fix HTTP client settings usage

### PR DESCRIPTION
**Pull request by jsumners**
**Description:** 

Previously, the `mezmoexporter` was ignoring any settings provided by way of https://github.com/open-telemetry/opentelemetry-collector/tree/4d445807f9eca3af660bec2a3fb4ca3518cd255d/config/confighttp. This PR fixes that mistake.

**Link to tracking Issue:** N/A

**Testing:**

**Documentation:**

### CODE REVIEW

**Comment by jsumners** [comment]
Is anyone able to take a look at this?

**Comment by jsumners** [comment]
All required CI is green 👍

**Comment by jsumners** [comment]
Is there anything I need to address before this can be merged?

**Comment by dashpole** [comment]
cc @billmeyer @gjanco

**Review by dashpole — APPROVED** [review]

**Comment by jsumners** [comment]
> cc @billmeyer @gjanco

@dashpole for reference, I am part of the same organization as those individuals. In fact, Greg Janco is my manager.

**Review by billmeyer — APPROVED** [review]

**Comment by dashpole** [comment]
@jsumners thanks.  I'll try to remember, but no promises ;).

I'm just the approver "sponsor" of the component, and while I can do reviews, I prefer to let the other codeowners from the component review before merging a PR.  If the existing codeowners want, you can be added as a codeowner as well.  But I generally rely on the codeowners files to tell me who I need to ping for reviews when I forget who works on what.

**Review by codeboten — APPROVED** [review]

**Comment by jsumners** [comment]
Thanks folks.

**Pull request merged**
